packages/perseus/src/widgets/free-response/serialize-free-response.test.ts

```diff
@@ -0,0 +1,151 @@
+ *
+ * This API is not built in a way that supports migrating data
+ * between versions of Perseus JSON. In fact serialization
+ * doesn't use WidgetOptions, but RenderProps; it's leveraging
+ * what is considered an internal implementation detail to support
+ * rehydrating previous state.
+ *
+ * The API is very fragile and likely broken. We have a ticket to remove it.
+ * However we don't have the bandwidth to implement an alternative right now,
+ * so I'm adding tests to make sure we're roughly still able to support
+ * what little we've been supporting so far.
+ *
+ * This API needs to be removed and these tests need to be removed with it.
+ */
+describe("FreeResponse serialization", () => {
+    function generateBasicFreeResponse(): PerseusItem {
+        const question = generateTestPerseusRenderer({
+            content: "[[☃ free-response 1]]",
+            widgets: {
+                "free-response 1": {
+                    type: "free-response",
+                    options: {
+                        allowUnlimitedCharacters: false,
+                        characterLimit: 500,
+                        placeholder: "test-placeholder",
+                        question: "test-question",
+                        scoringCriteria: [
+                            {
+                                text: "test-criterion",
+                            },
+                        ],
+                    },
+                },
+            },
+        });
+        const item = generateTestPerseusItem({question});
+        return item;
+    }
+
+    beforeAll(() => {
+        registerAllWidgetsForTesting();
+    });
+
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+
+        jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+            testDependencies,
+        );
+    });
+
+    afterEach(() => {
+        // The Renderer uses a timer to wait for widgets to complete rendering.
+        // If we don't spin the timers here, then the timer fires in the test
+        // _after_ and breaks it because we do setState() in the callback,
+        // and by that point the component has been unmounted.
+        act(() => jest.runOnlyPendingTimers());
+    });
+
+    it("should serialize the current state", async () => {
+        // Arrange
+        const {renderer} = renderQuestion(generateBasicFreeResponse());
+
+        await userEvent.type(
+            screen.getByLabelText("test-question"),
+            "test-answer",
+        );
+
+        // Act
+        const state = renderer.getSerializedState();
+        const userInput = renderer.getUserInput();
+
+        // Assert
+        // FreeResponse doesn't seem to include the user's answer
+        // in the serialized state, this assertion just affirms
+        // that there _is_ a user input, it's just not serialized
```

Comment:
Is it bad that the user's input isn't included in the object returned by `getSerializedState()`? That could easily be added.